### ISSUE TITLE
Node adapter sends `Vary: Accept-Encoding` even when not applicable

### ISSUE THREAD

**Issue by xamir82** [sveltejs/kit#10343]
### Describe the bug

The Node adapter currently sends back the `Vary: Accept-Encoding` header as part of the response (when files under `/_app` like `.js` and `.css` files are requested) even when the `precompress` configuration option is `false`; meaning even when the response **won't** _vary_ based on the `Accept-Encoding` header of the request, which is what `Vary: Accept-Encoding` is supposed to indicate.

In summary, `adapter-node` should not include `Vary: Accept-Encoding` in the response if `precompress: false`.

### Reproduction

Just build a SvelteKit Demo App with `adapter-node` and send a request using `curl -i` to some CSS/JS file, and examine the response headers.

### System Info

```Shell
-
```

### Severity

annoyance

**Comment by dummdidumm** [comment]
We don't set that header anywhere, which makes me think this is either from `polka` or sth else.

**Comment by xamir82** [comment]
> We don't set that header anywhere, which makes me think this is either from polka or sth else

Ah okay, then I guess yeah probably.

**Comment by theandrew168** [comment]
It looks like this header is actually being added by [sirv](https://github.com/lukeed/sirv/blob/1135207e92c40354543cbd15c763c7a61d79d432/packages/sirv/index.mjs#L191) (behavior added in [this PR](https://github.com/lukeed/sirv/pull/95)). SvelteKit utilizes sirv with both `gzip` and `brotli` unconditionally enabled [here](https://github.com/sveltejs/kit/blob/80ffb53382e397a8fc83e6b63d2675eeabb427bd/packages/adapter-node/src/handler.js#L50-L51). Looking at sirv's code, it looks like this header is added when `gzip` and / or `brotli` compression is generally enabled without also checking if a corresponding compressed file actually exists.

I see two potential solutions (though I'm unsure which would be better / more correct):
1. Only pass `gzip: true` and `brotli: true` to sirv when `precompress: true` is set
   1. This might cast too wide of a net and unintentionally impact other aspects of `adapter-node` that utilize asset compression
3. Fix sirv to only add `Vary: Accept-Encoding` when a compressed version of the requested file is present
   1. This seems like the "more correct" approach to me (it'd also solve this for any other users of sirv)

If a best approach can be agreed upon, I have the capacity to make either fix.

**Comment by teemingc** [comment]
Hi Andrew. I think both are correct based on the docs https://github.com/lukeed/sirv/tree/main/packages/sirv#optsgzip it says that the options should only be enabled for precompressed assets. So we should disable these if precompress is false.

It also says that it looks for .gzip files but I don’t think it’s doing that at the moment.

Feel free to submit a PR for both options! The Kit one might be easier to get done first
